slides.md

```diff
@@ -156,57 +156,97 @@ import anotherFunction, {aFunction} from "./someFile";
+```
+
+> Inside a component you can include a JS expression with `{expression}`, in this case `{todo.name}`.
+
+Since `todoElements` is an expression, we can include it inside our `View`, instead of the 3 `Text` elements, remember to use `{}`!
 
 ---
 # Keys in loops
-Components rendered with loops require a `key` property that identifies them.
-Keys are important for react to be able to understand which views have been added, removed or altered. Use `todo.name` as a key for now.
 
----
-# Conventions
-- Sometimes I'll reference React Native components, such as `Text`. These can be imported with `import { ComponentName } from "react-native";`.
-- When you need to create a new component, save it in `ComponentName.js` and import it with `import { ComponentName } from "./ComponentName';`.
+If you reloaded, you'll notice there's a small yellow warning on the bottom of the app. Components rendered with loops require a `key` property that identifies them.
+Keys are important for react to be able to understand which views have been added, removed or altered.
+
+> As you can do inside a component, you can use `propName={expression}` to pass properties. If it's a hardcoded string, prefer `propName="value"`.
+
+Before moving to the next step, let's use `todo.name` as a key for our elements.
 
 ---
 # Abstracting away
-The app looks amazing! But it would be great to have a component `Todo` instead of using `Text`. The simplest components can be functions that accept properties and return other components, strings or `null`. Pay attention to the brackets for the function, we'll go over that next.
+
+Now that our single todo is well isolated, we can extract it to a separate component.
+
+> The simplest component is a function that returns a tree of elements. It must have no side effects and return the same result given the same input.
+
 ```javascript
-export const MyComponent = ({name}) => <Text>Hello {name}!</Text>;
-<MyComponent name="Daniele" /> // Somewhere in the app
+const Hello = ({name}) => (
+  <Text>Hello {name}!</Text>
+);
 ```
-You can reference JS variables using `{curlyBraces}`.
 
----
-# ({a, b, c})?!?
-To write an anonymous function in JS, you can use `const add = (p2, p2) => { return p1 + p2; }`. There are some tricks to make it easier to write though.
-- Instead of having the body, you can directly return an expression `(p1, p2) => p1 + p2`
-- If your parameter is an object, you can directly reference its keys with `{k1, k2}`.
-- React Components receive an object called `properties`, containing the attributes we pass to the component when we use it: `<MyComponent name="Daniele" />`.
+> The component will receive a parameter containing the props passed to it, in this example `name` is one of the props `Hello` can accept, and we're using destructuring to access it directly.
+
+> You can then use the component importing it (if it was declared it in another file, remember to export it!) and using it as any other component.
```

Comment:
I think this could be rewritten, it's hard to keep track of each "it"

Reply:
20324b137f36a20c541a495c5b61fd75acc11418